keanu-project/src/main/java/io/improbable/keanu/algorithms/Samples.java

```diff
@@ -52,4 +53,8 @@ public T getMode() {
     public List<T> asList() {
         return new ArrayList<>(samples);
     }
+
+    public T asTensor() {
+        throw new NotImplementedException("Generic samples cannot be coerced into a tensor.");
```

Comment:
I would prefer `UnsupportedOperationException`. It indicates that this will not ever be supported, whereas "not implemented" is usually interpreted as "not _yet_ implemented". Also, it's part of the standard library.

see also https://stackoverflow.com/a/22045192/741789